### PR TITLE
Feature: Add site data to placement DTO

### PR DESCRIPTION
**Pull request by ReubenRobertsHEE**
Somewhat speculative pull request. It passes the new tests, but I'd like to trial-load some actual data to see that the site data is carried across into the placement.

### CODE REVIEW

File: src/main/java/uk/nhs/hee/tis/trainee/sync/facade/PlacementEnricherFacade.java [hunk 1]
```diff
@@ -179,6 +211,73 @@ private void syncPlacement(Placement placement, String employingBodyName,
...
+      placementData.put(PLACEMENT_DATA_SITE_LOCATION, siteLocation);
+    }
+
+    // Set the required metadata so the record can be synced using common logic.
+    placement.setOperation(LOAD);
+    placement.setSchema("tcs");
+    placement.setTable("Placement");
+
+    tcsSyncService.syncRecord(placement);
+  }
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/sync/facade/PlacementEnricherFacade.java:239]: This needs building in to `syncPlacementDetails`, everything must be synced together after building the complete record.

**Review by Judge40 — CHANGES_REQUESTED** [review]
Getting a site will need to trigger the other enrichments as well (Post/Trust).

We might need to consider how best to do that, maybe having each path add its data to the placement and then calling the `TcsSyncService` in a separate step would make more sense than tying site and trust together too closely.

Also missing some updates to `TraineeDetailsDto` and its associated mapper to add the site fields.

File: src/main/java/uk/nhs/hee/tis/trainee/sync/facade/PlacementEnricherFacade.java (hunk 1 above)
Comment (ReubenRobertsHEE) [reply src/main/java/uk/nhs/hee/tis/trainee/sync/facade/PlacementEnricherFacade.java:239]: Thanks - yes, I was wondering about that! :)

File: build.gradle [hunk 2]
```diff
@@ -47,6 +47,9 @@ dependencies {
 
   // Required to support PATCH requests.
   implementation "org.apache.httpcomponents:httpclient:4.5.12"
+
+  // make When enum available
+  implementation 'com.google.code.findbugs:jsr305:3.0.2'
```
[the remark this answers is not in the document]
Comment (ReubenRobertsHEE) [reply build.gradle:52]: Ah, that's not essential. When I build/run I was getting a lot of these warnings:

warning: unknown enum constant When.MAYBE
  reason: class file for javax.annotation.meta.When not found
  
  Adding that dependency makes them go away (as per https://stackoverflow.com/questions/53326271/spring-nullable-annotation-generates-unknown-enum-constant-warning ), but its easy to remove it if you prefer.

[the remark this answers is not in the document]
Comment (Judge40) [reply build.gradle:52]: I'd rather deal with (ignore) the warning than add another dependency to the build.

[the remark this answers is not in the document]
Comment (ReubenRobertsHEE) [reply build.gradle]: Sure thing - removed

**Review by Judge40 — APPROVED** [review]

**Pull request merged**